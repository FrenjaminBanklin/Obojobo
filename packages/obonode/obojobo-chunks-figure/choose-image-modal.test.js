import React from 'react'
import { mount } from 'enzyme'

import ChooseImageModal from './choose-image-modal'

jest.mock('obojobo-document-engine/src/scripts/viewer/util/api-util', () => ({
	postMultiPart: jest.fn().mockResolvedValue({ mediaId: 'mockMediaId' }),
	get: jest.fn().mockResolvedValue({
		json: jest.fn().mockResolvedValue({ data: [{ id: '0' }], hasMore: true })
	})
}))

describe('Image Properties Modal', () => {
	beforeEach(() => {
		jest.clearAllMocks()
	})

	test('ChooseImageModal component', () => {
		const component = mount(<ChooseImageModal />)
		const tree = component.html()

		expect(tree).toMatchSnapshot()
	})

	test('ImageProperties component changes url', () => {
		const component = mount(<ChooseImageModal onConfirm={jest.fn} />)

		component
			.find('#choose-image--image-controls--url')
			.simulate('change', { target: { value: 'changed url' } })

		const instance = component.instance()
		expect(instance.state.url).toBe('changed url')
	})

	test('ImageProperties component changes file', () => {
		const component = mount(
			<ChooseImageModal onSetMediaUrl={jest.fn} onSetIsChoosingImage={jest.fn} />
		)

		component.find('#choose-image--image-controls--upload').simulate('change', {
			target: {
				value: 'changed',
				files: [
					new window.Blob([JSON.stringify({ name: 'mockFileName' })], { type: 'application/json' })
				]
			}
		})

		expect(component.html()).toMatchSnapshot()
	})

	test('ImageProperties click on `View More...`', () => {
		const component = mount(
			<ChooseImageModal onSetIsChoosingImage={jest.fn} onSetMediaUrl={jest.fn} />
		)

		component.setState({ medias: [{ id: 'mock_id' }], isFetching: false, hasMore: true })

		expect(component.instance().state.hasMore).toBe(true)

		component.find('.choose-image--image-gallary--view-more-btn').simulate('click')

		expect(component.html()).toMatchSnapshot()
	})

	test('ImageProperties click on an image', () => {
		const onSetMediaUrl = jest.fn()
		const onSetIsChoosingImage = jest.fn()
		const component = mount(
			<ChooseImageModal onSetIsChoosingImage={onSetIsChoosingImage} onSetMediaUrl={onSetMediaUrl} />
		)

		component.setState({ medias: [{ id: 'mock_id' }], isFetching: false, hasMore: false })
		component.find('.image-gallary--single-photo').simulate('click')

		expect(onSetMediaUrl).toHaveBeenCalled()
		expect(onSetIsChoosingImage).toHaveBeenCalled()

		expect(component.html()).toMatchSnapshot()
	})

	test('ImageProperties handle fetching error', () => {
		const APIUtil = require('obojobo-document-engine/src/scripts/viewer/util/api-util')
		APIUtil.get = jest.fn().mockResolvedValue({
			json: jest.fn().mockResolvedValue({ status: 'error' })
		})

		const component = mount(
			<ChooseImageModal onSetIsChoosingImage={jest.fn} onSetMediaUrl={jest.fn} />
		)

		expect(component.html()).toMatchSnapshot()
	})
})
