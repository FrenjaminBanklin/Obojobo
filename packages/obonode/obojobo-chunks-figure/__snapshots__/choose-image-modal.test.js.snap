// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose Image Modal ChooseImageModal component 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><h5>Loading...</h5></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ChooseImageModal component focuses on first element 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><h5>Loading...</h5></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties click on \`Cancel\` 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><h5>Loading...</h5></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties click on \`OK\` 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><h5>Loading...</h5></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties click on \`View More...\` 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><img tabindex=\\"0\\" class=\\"image-gallary--single-photo\\" id=\\"mock_id\\" src=\\"/api/media/mock_id/small\\"><h5>Loading...</h5></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties click on an image 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><img tabindex=\\"0\\" class=\\"image-gallary--single-photo\\" id=\\"mock_id\\" src=\\"/api/media/mock_id/small\\"></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties component changes file 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><h5>Loading...</h5></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties handle fetching error 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><h5>Loading...</h5></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties onKeyPress \`enter\` on an image 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><img tabindex=\\"0\\" class=\\"image-gallary--single-photo\\" id=\\"mock_id\\" src=\\"/api/media/mock_id/small\\"></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Choose Image Modal ImageProperties onPress on an image 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">Choose an image</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"choose-image\\"><div class=\\"choose-image--image-controls\\"><div class=\\"choose-image--image-controls--upload\\"><input tabindex=\\"0\\" type=\\"file\\" name=\\"file\\" id=\\"file\\" class=\\"inputfile\\"><label for=\\"file\\">Upload new image</label></div><div class=\\"choose-image--image-controls--or\\">or</div><input id=\\"choose-image--image-controls--url\\" type=\\"text\\" placeholder=\\"Enter URL\\" value=\\"\\"></div><div class=\\"choose-image--divider\\"></div><small>Recently uploaded</small><div class=\\"choose-image--image-gallary\\"><img tabindex=\\"0\\" class=\\"image-gallary--single-photo\\" id=\\"mock_id\\" src=\\"/api/media/mock_id/small\\"></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;
